src/parameterized_tendencies/gravity_wave_drag/non_orographic_gravity_wave.jl

```diff
@@ -242,33 +298,35 @@ function non_orographic_gravity_wave_tendency!(
 end
 
 function non_orographic_gravity_wave_forcing(
-    ᶜu,
-    ᶜbf,
-    ᶜρ,
-    ᶜz,
+    old_ᶜu,
+    old_ᶜbf,
+    old_ᶜρ,
+    old_ᶜz,
     source_level,
     damp_level,
     source_ampl,
     Bw,
     Bn,
+    B0,
     cw,
     cn,
     flag,
     c,
     c0,
     nk,
 )
-    FT = eltype(ᶜz)
+    FT = eltype(old_ᶜz)
     # add an extra layer above model top so that forcing between the very top
     # model layer and the upper boundary can be calculated
-    append!(ᶜu, FT(2) * ᶜu[end] - ᶜu[end - 1])
-    append!(ᶜρ, ᶜρ[end] * ᶜρ[end] / ᶜρ[end - 1])
-    append!(ᶜbf, ᶜbf[end])
-    append!(ᶜz, FT(2) * ᶜz[end] - ᶜz[end - 1])
+    ᶜu = vcat(old_ᶜu, FT(2) * old_ᶜu[end] - old_ᶜu[end - 1])
```

Comment:
Just to note, `vcat` is not really fixing the issue here, but the change is fine.